dsp/window/cmd/leakage/leakage.go

```diff
@@ -0,0 +1,384 @@
+	m := flag.Int("m", 1000, "specify sample length (must be greater than n)")
+	csv := flag.String("csv", "", "specify an input file of frequency amplitudes (dB transformed)")
+	out := flag.String("o", "", "specify output file for plot")
+	comment := flag.Bool("comment", false, "output a comment line for the window (only for window function)")
+	flag.Parse()
+
+	win := windows[strings.ToLower(*name)]
+	if win == nil && *csv == "" {
+		fmt.Fprintf(os.Stderr, "missing function name or csv input")
+		flag.Usage()
+		os.Exit(2)
+	}
+	if *csv == "" && !win.ok(*param) {
+		fmt.Fprintf(os.Stderr, "missing parameter")
+		flag.Usage()
+		os.Exit(2)
+	}
+	if *out == "" {
+		fmt.Fprintf(os.Stderr, "missing output filename")
+		flag.Usage()
+		os.Exit(2)
+	}
+
+	p := plot.New()
+	p.X.Label.Text = "DFT bin"
+	p.Y.Label.Text = "Amplitude [dB]"
+	p.Add(plotter.NewGrid())
+
+	var (
+		c        *characteristics
+		spectrum plotter.XYs
+		min      float64
+		err      error
+	)
+	if win != nil {
+		symmetry := "(symmetric)"
+		if !*symm {
+			symmetry = "(periodic)"
+		}
+		p.Title.Text = fmt.Sprintf("Spectral Leakage for %s %s", win.name(*param), symmetry)
+		c, spectrum, min, err = funcCharacteristics(win.fn(*param), *n, *m, *symm)
+		if err != nil {
+			log.Fatal(err)
+		}
+		if *comment {
+			fmt.Printf("// Spectral leakage parameters: ΔF_0 = %2f, ΔF_0.5 = %.2f, K = %.2f, ɣ_max = %2f, β = %2f.\n",
+				c.deltaF0, c.deltaFhalf, c.k(), c.gammaMax, c.beta)
+		}
+	} else {
+		f, err := os.Open(*csv)
+		if err != nil {
+			log.Fatal(err)
+		}
+		defer f.Close()
+		p.Title.Text = fmt.Sprintf("Spectral Leakage for %s", *csv)
+		c, spectrum, min, err = csvCharacteristics(f, *n, *m)
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+
+	spectrumLines, err := plotter.NewLine(spectrum)
+	if err != nil {
+		log.Fatalf("spectrum: %v", err)
+	}
+
+	gammaLine, err := plotter.NewLine(plotter.XYs{
+		{X: 0, Y: c.gammaMax}, {X: float64(*n) / 2, Y: c.gammaMax},
+	})
+	if err != nil {
+		log.Fatalf("ɣ_max: %v", err)
+	}
+	gammaLine.Color = color.RGBA{R: 0x40, G: 0x80, B: 0xff, A: 0xff}
+
+	deltaF0Line, err := plotter.NewLine(plotter.XYs{
+		{X: c.deltaF0 / 2, Y: 0}, {X: c.deltaF0 / 2, Y: min},
+	})
+	if err != nil {
+		log.Fatalf("ΔF_0: %v", err)
+	}
+	deltaF0Line.Color = color.RGBA{R: 0xff, A: 0xff}
+
+	deltaFhalfLine, err := plotter.NewLine(plotter.XYs{
+		{X: c.deltaFhalf / 2, Y: 0}, {X: c.deltaFhalf / 2, Y: min},
+	})
+	if err != nil {
+		log.Fatalf("ΔF_0.5: %v", err)
+	}
+	deltaFhalfLine.Color = color.RGBA{G: 0xff, A: 0xff}
+
+	var blank plotter.Line
+
+	p.Add(
+		gammaLine,
+		deltaF0Line,
+		deltaFhalfLine,
+		spectrumLines,
+	)
+	p.Legend.Add(fmt.Sprintf("ΔF_0 = %.3v", c.deltaF0), deltaF0Line)
+	p.Legend.Add(fmt.Sprintf("ΔF_0.5 = %.3v", c.deltaFhalf), deltaFhalfLine)
+	p.Legend.Add(fmt.Sprintf("K = %.3v", c.k()), &blank)
+	p.Legend.Add(fmt.Sprintf("ɣ_max = %.3v", c.gammaMax), gammaLine)
+	p.Legend.Add(fmt.Sprintf("β = %.3v", c.beta), &blank)
```

Comment:
@sbinet I had difficulties getting the LaTeX equivalent of this working. when I replace these with the following and have the needed imports and setup I get a panic.
```
	p.Legend.Add(fmt.Sprintf(`${\Delta}F_0 = %.3v$`, c.deltaF0), deltaF0Line)
	p.Legend.Add(fmt.Sprintf(`${\Delta}F_{0.5} = %.3v$`, c.deltaFhalf), deltaFhalfLine)
	p.Legend.Add(fmt.Sprintf(`$K = %.3v$`, c.k()), &blank)
	p.Legend.Add(fmt.Sprintf(`${\gamma}_{max} = %.3v$`, c.gammaMax), gammaLine)
	p.Legend.Add(fmt.Sprintf(`$\beta = %.3v$`, c.beta), &blank)
```
The prefix of the panic is
```

goroutine 1 [running]:
github.com/go-latex/latex/mtex.(*visitor).Visit(0xc0001720e0, 0x7223d8, 0xc000126558, 0x0, 0x0)
	/home/daniel/pkg/mod/github.com/go-latex/latex@v0.0.0-20210118124228-b3d85cf34e07/mtex/parser.go:128 +0xec8
<snip>
```
This appears to be related to the subscript underscore.

Reply:
Also, the line spacing with the LaTeX handling looks pretty tight. This is probably related to the issues you filed wrt font description parameters.

Here is the non-LaTeX output

![sine_20_1000_go_nolatex](https://user-images.githubusercontent.com/275221/112391771-52212e00-8d48-11eb-8605-8694a4c43d84.png)

and here is the LaTeX-using output (with removal of {} wrapping which swallows the text and the subscripts which panic).

![sine_20_1000_go](https://user-images.githubusercontent.com/275221/112391806-6107e080-8d48-11eb-9323-85ef5f72b03b.png)

Reply:
yeah, the handling (or lack thereof) of subscripts/superscripts is a known limitation of `go-latex`.
(that's partly whay I've started `star-tex.org/x/tex`)
- https://github.com/go-latex/latex/issues/17

as for the lack of an adequate linegap for the legend, that's news to me. (but may very well be related to https://github.com/gonum/plot/issues/676)

Reply:
Yeah, I wen't looking and then saw that issue. The other thing that I noticed is the `{text}` doesn't end up rendering "text". I think it should.